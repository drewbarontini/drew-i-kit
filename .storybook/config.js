import { configure, addDecorator } from '@storybook/react';
import React, { Fragment } from 'react';
import { ThemeProvider } from 'styled-components';
import 'normalize.css';

import { theme } from '../config';

import GlobalStyles from '../components/GlobalStyles';

const req = require.context('../components/', true, /.story.js$/);
function loadStories() {
  req.keys().forEach(filename => req(filename));
}

function withStyledComponents(story) {
  return (
    <ThemeProvider theme={theme}>
      <Fragment>
        {story()}
        <GlobalStyles />
      </Fragment>
    </ThemeProvider>
  );
}

addDecorator(withStyledComponents);
configure(loadStories, module);
