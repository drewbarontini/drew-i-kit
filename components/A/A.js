import PropTypes from 'prop-types';
import styled, { css } from 'styled-components';
import { ifNotProp } from 'styled-tools';

import { hover, palette, setProp } from '../../lib/styleUtils';

const A = styled.a`
  color: ${palette('primary')};
  text-decoration: none;

  ${setProp({
    prop: 'color',
    themeKey: 'colors',
  })};
  ${setProp({
    prop: 'fontSize',
    cssProp: 'font-size',
    themeKey: 'fonts.sizes',
  })};
  ${setProp({
    prop: 'textDecoration',
    cssProp: 'text-decoration',
  })};

  /*
    NOTE: This is used because styled-components has an issue when 
    this is using React Router's Link component because it will pass
    down the prop all the way to the DOM where a warning is thrown.
  */
  ${hover(css`
    color: ${props =>
      props['data-hover-color']
        ? palette(props['data-hover-color'])
        : palette('fg')};
  `)};
`;

A.propTypes = {
  color: PropTypes.string,
  fontSize: PropTypes.string,
  textDecoration: PropTypes.string,
  'data-hover-color': PropTypes.string,
};

A.displayName = 'A';

export default A;
