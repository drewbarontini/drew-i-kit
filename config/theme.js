const theme = {
  borders: {
    radius: {
      base: '3px',
    },
  },
  colors: {
    bg: '#fff',
    fg: '#272727',
    primary: '#3576a7',
    subdued: '#cbcbcb',
    error: '#fe5f55',
    success: '#8acb88',
    warning: '#ffc527',
  },
  fonts: {
    family: {
      base:
        "-apple-system, BlinkMacSystemFont, 'Segoe UI', Roboto, Oxygen-Sans, Ubuntu, Cantarell, 'Helvetica Neue', sans-serif",
      heading:
        "-apple-system, BlinkMacSystemFont, 'Segoe UI', Roboto, Oxygen-Sans, Ubuntu, Cantarell, 'Helvetica Neue', sans-serif",
    },
    lineHeight: {
      base: '1.5',
      heading: '1.2',
    },
    sizes: {
      base: '14px',
      xxs: '8px',
      xs: '10px',
      s: '12px',
      l: '16px',
      xl: '24px',
      h1: '26px',
    },
    weights: {
      light: 200,
      normal: 400,
      semiBold: 600,
      bold: 800,
    },
  },
  spacing: {
    base: '16px',
    xs: '5px',
    s: '10px',
    m: '12px',
    l: '32px',
    xl: '60px',
  },
};

export default theme;
