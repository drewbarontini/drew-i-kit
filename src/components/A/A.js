import PropTypes from 'prop-types';
import styled, { css } from 'styled-components';

import { hover } from '../../lib/hover';
import { palette, setProp, typography } from '../../lib/props';
import { responsive } from '../../lib/responsive';

const A = styled.a`
  color: ${palette('primary.base')};
  text-decoration: none;

  ${setProp({
    prop: 'color',
    themeKey: 'colors',
  })};
  ${setProp({
    prop: 'textDecoration',
  })};

  /*
    NOTE: This is used because styled-components has an issue when 
    this is using React Router's Link component because it will pass
    down the prop all the way to the DOM where a warning is thrown.
  */
  ${hover(css`
    color: ${props =>
      props['data-hover-color']
        ? palette(props['data-hover-color'])
        : palette('fg')};
  `)};

  ${typography}
  ${responsive}
`;

A.propTypes = {
  color: PropTypes.string,
  fontSize: PropTypes.string,
  fontStyle: PropTypes.string,
  fontWeight: PropTypes.string,
  lineHeight: PropTypes.string,
  textAlign: PropTypes.string,
  textDecoration: PropTypes.string,
  'data-hover-color': PropTypes.string,
};

A.displayName = 'A';

export default A;
