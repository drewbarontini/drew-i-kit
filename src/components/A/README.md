# A

### Usage

```jsx
import { A } from 'drew-i-kit';

<A href="#0">My Link</A>
```

```jsx
import { A } from 'drew-i-kit';

<A href="#0" color="neutral.base">My Link</A>
<A href="#0" fontSize="l">My Link</A>
<A href="#0" textDecoration="underline">My Link</A>
<A href="#0" data-hover-color="neutral.base">My Link</A>
```

### Props

- `color` (**string**, optional)
- `fontSize` (**string**, optional)
- `textDecoration` (**string**, optional)
- `data-hover-color` (**string**, optional)