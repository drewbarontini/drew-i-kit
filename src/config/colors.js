import { paletteFor } from '../lib';

const colors = {
  ...paletteFor({
    area: 'primary',
    base: '#3576a7',
    darker: ['#152f43', '#1d415c', '#255375', '#2d648e'],
    lighter: ['#538bb4', '#729fc1', '#90B4cf', '#aeC8dc'],
  }),
  ...paletteFor({
    area: 'neutral',
    base: '#737373',
    darker: ['#0b0b0b', '#1d1d1d', '#2e2e2e', '#3f3f3f', '#515151', '#626262'],
    lighter: ['#888888', '#9d9d9d', '#b2b2b2', '#c7c7c7', '#dcdcdc', '#f1f1f1'],
  }),
  accent: {
    ...paletteFor({
      area: 'green',
      base: '#8acb88',
      darker: ['#233322', '#375136', '#4c704b', '#618e5f', '#75ad74'],
      lighter: ['#9cd39a', '#addbac', '#bfe2be', '#d0eacf', '#e2f2e1'],
    }),
    ...paletteFor({
      area: 'yellow',
      base: '#ffc527',
      darker: ['#40310a', '#664f10', '#8c6c15', '#b38a1b', '#d9a721'],
      lighter: ['#ffce47', '#ffd668', '#ffdf88', '#ffe8a9', '#fff1c9'],
    }),
    ...paletteFor({
      area: 'red',
      base: '#fe5f55',
      darker: ['#401815', '#662622', '#8c342f', '#b2433b', '#d85148'],
      lighter: ['#fe776f', '#fe8f88', '#fea7a2', '#ffbfbb', '#ffd7d5'],
    }),
  },
  bg: '#fff',
  fg: '#272727',
  white: '#fff',
  black: '#272727',
  border: '#dcdcdc',
};

export default colors;
