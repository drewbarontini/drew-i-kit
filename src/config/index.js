export { default as cssProps } from './cssProps';
export { default as icons } from './icons';
export { default as theme } from './theme';
