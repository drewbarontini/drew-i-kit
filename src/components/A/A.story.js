import React from 'react';
import { storiesOf } from '@storybook/react';

import A from '.';

storiesOf('A', module)
  .add('default', () => <A href="#0">Link</A>)
  .add('color', () => (
    <A href="#0" color="neutral.base">
      Link
    </A>
  ))
  .add('font size', () => (
    <A href="#0" fontSize="l">
      Link
    </A>
  ))
  .add('text decoration', () => (
    <A href="#0" textDecoration="underline">
      Link
    </A>
  ))
  .add('hover color', () => (
    <A href="#0" data-hover-color="neutral.base">
      Link
    </A>
  ))
  .add('responsive', () => (
    <A
      href="#0"
      breakpoints={[
        {
          size: 's',
          styles: theme => `color: ${theme.colors.accent.red.base};`,
        },
        {
          size: 'm',
          styles: theme => `color: ${theme.colors.accent.green.base};`,
        },
      ]}
    >
      Link
    </A>
  ));
