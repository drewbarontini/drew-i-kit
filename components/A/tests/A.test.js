import React from 'react';
import { render } from 'react-testing-library';
import 'jest-styled-components';

import { theme } from '../../../config';

import A from '../';

describe('<A />', () => {
  it('should render properly', () => {
    const { container } = render(
      <A href="#0" theme={theme}>
        My Link
      </A>
    );
    expect(container.firstChild).toMatchSnapshot();
    expect(container.firstChild).toHaveStyleRule('color', theme.colors.primary);
    expect(container.firstChild).toHaveStyleRule('text-decoration', 'none');
  });
});
