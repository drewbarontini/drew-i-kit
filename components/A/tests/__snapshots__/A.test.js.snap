// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<A /> should render properly 1`] = `
.c0 {
  color: #3576a7;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c0:active,
.c0:focus,
.c0:hover {
  color: #272727;
}

<a
  class="c0"
  href="#0"
>
  My Link
</a>
`;
